src/modules/rewards/services/opRebateServiceV2.ts

```diff
@@ -0,0 +1,347 @@
+        .getRawOne<{
+          volumeUsd: number;
+        }>(),
+      // TODO: add claimable rewards
+    ]);
+
+    return {
+      depositsCount: parseInt(depositsCount),
+      unclaimedRewards,
+      volumeUsd,
+      claimableRewards: "0",
+    };
+  }
+
+  // public async getOpRebateRewards(query: GetRewardsQuery) {
+  //   const limit = parseInt(query.limit ?? "10");
+  //   const offset = parseInt(query.offset ?? "0");
+  //   const userAddress = assertValidAddress(query.userAddress);
+
+  //   const baseQuery = this.buildBaseQuery(this.rewardRepository.createQueryBuilder("r"), userAddress);
+
+  //   const rewardsQuery = baseQuery.orderBy("r.depositDate", "DESC").limit(limit).offset(offset);
+  //   const [rewards, total] = await rewardsQuery.getManyAndCount();
+     
+  //   // JOIN instead of query deposits separately 
+  //   const depositPrimaryKeys = rewards.map((reward) => reward.depositPrimaryKey);
+  //   const deposits = await this.depositRepository.find({
+  //     where: { id: In(depositPrimaryKeys) },
+  //   });
+
+  //   return {
+  //     rewards: rewards.map((reward) => ({
+  //       ...reward,
+  //       deposit: deposits.find((deposit) => deposit.id === reward.depositPrimaryKey),
+  //     })),
+  //     pagination: {
+  //       limit,
+  //       offset,
+  //       total,
+  //     },
+  //   };
+  // }
+
+  public async getOpRebateRewardsForDepositPrimaryKeys(depositPrimaryKeys: number[]) {
+    if (depositPrimaryKeys.length === 0) {
+      return [];
+    }
+    const rewardsQuery = this.rewardRepository
+      .createQueryBuilder("r")
+      .where("r.depositPrimaryKey IN (:...depositPrimaryKeys)", { depositPrimaryKeys })
+      .leftJoinAndSelect("r.rewardToken", "rewardToken");
+    const rewards = await rewardsQuery.getMany();
+    return rewards;
+  }
+
+  public async createOpRebatesForDeposit(depositId: number, originChainId: number) {
+    if (!this.appConfig.values.rewardPrograms["op-rebates"].enabled) {
+      this.logger.verbose(`OP rebate rewards are disabled. Skipping...`);
+      return;
+    }
+
+    const deposit: RewardedDeposit = await this.depositRepository.findOne({
+      where: { depositId, originChainId },
+    });
+
+    if (!deposit || !deposit.fillTxHash) return;
+    if (!this.isDepositEligibleForOpRewards(deposit)) return;
+    if (!deposit.totalBridgeFeeUsd) {
+      throw new Error(`Missing total bridge fee in USD (depositId: ${depositId}, originChainId: ${originChainId}).`);
+    }
+    if (!this.isDepositTimeAfterStart(deposit)) return;
+    if (!this.isDepositTimeBeforeEnd(deposit)) return;
+
+    // We use the `from` address of the deposit transaction as the reward receiver
+    // to also take into accounts deposits routed through the SpokePoolVerifier contract.
+    const provider = this.ethProvidersService.getProvider(deposit.originChainId);
+    const depositTransaction = await provider.getTransaction(deposit.depositTxHash); // This could also be part of the indexer data
```

Comment:
Maybe we could fetch this in the indexer as we are already retrieving the transaction.